0-module-and-module-source.emu

```diff
@@ -0,0 +1,661 @@
+<!doctype html>
+<html lang="en-GB-oxendict">
+<head>
+<meta charset="utf8">
+<link href="ecmarkup.css" rel="stylesheet">
+<script src="ecmarkup.js"></script>
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/solarized-light.min.css">
+</head>
+
+<body>
+<pre class="metadata">
+title: 'Module & ModuleSource API'
+stage: 1
+contributors: Mark Miller, Caridy Patiño, Patrick Soquet, Kris Kowal, Jack Works, Guy Bedford, Nicolò Ribaudo
+status: proposal
+copyright: false
+location: https://tc39.es/proposal-compartments/
+</pre>
+
+<emu-clause id="sec-well-known-intrinsic-objects">
+    <h1>Well-known intrinsic objects</h1>
+    <emu-table id="table-7" caption="Well-known Intrinsic Objects">
+        <table>
+            <tbody>
+                <tr>
+                    <th>Intrinsic Name</th>
+                    <th>Global Name</th>
+                    <th>ECMAScript Language Association</th>
+                </tr>
+                <tr>
+                    <td>%ModuleSource%</td>
+                    <td>`ModuleSource`</td>
+                    <td>The ModuleSource constructor (<emu-xref href="#sec-modulesource-constructor"></emu-xref>)</td>
+                </tr>
+                <tr>
+                    <td>%Module%</td>
+                    <td>`Module`</td>
+                    <td>The Module constructor (<emu-xref href="#sec-module-constructor"></emu-xref>)</td>
+                </tr>
+            </tbody>
+        </table>
+    </emu-table>
+</emu-clause>
+
+<emu-clause id="sec-source-text-module-records">
+    <h1>Source Text Module Records</h1>
+
+    <p>In addition to the fields defined in <emu-xref href="#table-cyclic-module-fields"></emu-xref>, Source Text Module Records have the additional fields listed in <emu-xref href="#table-additional-fields-of-source-text-module-records"></emu-xref>. Each of these fields is initially set in ParseModule.</p>
+    <emu-table id="table-additional-fields-of-source-text-module-records" caption="Additional Fields of Source Text Module Records" oldids="table-38">
+      <table>
+        <tr>
+          <th>
+            Field Name
+          </th>
+          <th>
+            Value Type
+          </th>
+          <th>
+            Meaning
+          </th>
+        </tr>
+        <tr>
+          <td>
+            ...
+          </td>
+          <td>
+          </td>
+          <td>
+          </td>
+        </tr>
+        <tr>
+          <td>
+            <ins>[[ModuleInstance]]</ins>
+          </td>
+          <td>
+            An instance of a %Module% constructor or *undefined*.
+          </td>
+          <td>
+            A Module Instance or *undefined* if no Module Instance is associated to this Record.
+          </td>
+        </tr>
+      </table>
+    </emu-table>
+</emu-clause>
+
+<emu-clause id="sec-modulesource-records">
+    <h1>Module Source Records</h1>
```

Comment:
We can do that later. The idea is to modify the minimum of the current spec to expedite layer 0. So, it is intentional. I don't think it belongs to the Cyclic hierarchy because this is not really a module record, this is just static parsing information. Maybe it should be just an internal slot of the Module Record.

Reply:
Sure that's fine for now.